presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationSmokeTest.java

```diff
@@ -116,6 +121,148 @@ private KafkaTopicFieldDescription createOneFieldDescription(String name, Type t
         return new KafkaTopicFieldDescription(name, type, mapping, null, dataFormat, null, false);
     }
 
+    @Test(dataProvider = "testRoundTripAllFormatsDataProvider")
+    public void testRoundTripAllFormats(RoundTripTestSetup roundTripTestSetup)
+    {
+        String tableName = roundTripTestSetup.getTableName();
+        String fieldNames = roundTripTestSetup.getFieldNames();
+        String fieldValues = roundTripTestSetup.getFieldValues();
+        String comparableName = roundTripTestSetup.getFieldName("f_bigint");
+        Object comparableValue = roundTripTestSetup.getFieldValue("f_bigint");
+
+        assertUpdate("INSERT into write_test." + tableName + " (" + fieldNames + ") VALUES (" + fieldValues + ")", 1);
+        assertQuery("SELECT " + fieldNames + " FROM write_test." + tableName + " WHERE " + comparableName + " = " + comparableValue, "VALUES (" + fieldValues + ")");
+    }
+
+    @DataProvider(name = "testRoundTripAllFormatsDataProvider")
+    public final Object[][] testRoundTripAllFormatsDataProvider()
+    {
+        return testRoundTripAllFormatsData().stream()
+                .map(roundTripTestSetup -> new Object[] {roundTripTestSetup})
+                .toArray(Object[][]::new);
+    }
+
+    private List<RoundTripTestSetup> testRoundTripAllFormatsData()
+    {
+        return ImmutableList.<TestKafkaIntegrationSmokeTest.RoundTripTestSetup>builder()
+                .build();
+    }
+
+    protected static final class RoundTripTestSetup
+    {
```

Comment:
The best would be have simple use case already implemented (possibly in separate commit) to see that all of this actually work.

Reply:
I have one for each format when I open encoder prs